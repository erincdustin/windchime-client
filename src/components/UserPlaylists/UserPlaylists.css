.result {
  background-color: rgb(232, 234, 235);
  border-radius: 10px;
  width: 100%;
  margin: 10px auto;
  padding-bottom: 25px;
  text-align: center;
  display: inline-block;
  min-width: 275px;
}

p {
  font-size: 12px;
  display: inline;
  padding: 0px 5px;
}

.banner {
  background-color: rgb(77, 153, 167); 
  border-top-left-radius: 10px;
  border-top-right-radius: 10px;
  margin-bottom: 15px;
  height: 25px;
  width: 100%;
  color: white;
}

.center {
  display: inline-block;
  text-align: center;
  margin: 0px auto;
}

.iframe{
  margin: auto;
  text-align: center;
  display: block;
}

.rating{
  height: 10px;
  margin: 2px 1px;
}

.result p{
  display: block;
  text-align: left;
  padding-left: 12.5%;
}

.playlist-center{
  text-align: center;
}

.none{
  margin: 25px auto;
  font-weight: bold;
  text-align: center;
}

.playlist-center h4{
  font-weight: bold;
  margin: 10px auto;
}

.playlist-center p{
  display: block;
  margin: 10px auto;
}

.no-playlists{
  width: 80%;
  text-align: center;
  margin: auto;
}

.playlist-center .btn{
 width: 200px;
 margin: 10px 5px;
}

@media screen and (min-width: 450px) {
  p {
    font-size: 14px;
  }
}

@media screen and (min-width: 750px) {

  .playlist-center{
    width: 100%;
  }

  .flex{
    display: flex;
    flex-wrap: wrap;
    width: 100%;
  }

  .result{
      background-color: rgb(232, 234, 235);
      border-radius: 10px;
      margin: 10px 2%;
      padding-bottom: 25px;
      text-align: center;
      flex: 45%;
  }
}
