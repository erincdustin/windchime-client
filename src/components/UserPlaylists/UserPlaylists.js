import React from 'react';
import { Link } from 'react-router-dom';
import Playlist from '../Playlist/Playlist';
import config from '../../config'
import { format } from 'date-fns'
import './UserPlaylists.css';
import TokenService from '../../services/token-service';

class UserPlaylists extends React.Component {

  state = {
    userPlaylists: [],
  }

  componentDidMount() {
    const URL = `${config.API_ENDPOINT}/playlists`;

    return fetch(URL)
    .then(res => (!res.ok)
    ? res.json().then(e => Promise.reject(e))
    : res.json()
)
    .then(res=> {
      this.setState({ userPlaylists: res })
  })
}

  render() {
  const mappedPlaylists = this.state.userPlaylists.filter((playlist) => playlist.user_id === this.props.id).map((playlist, index) => {
    return (
     <div className="result" key={playlist.playlist_id}>
       <div class="banner">Playlist {index + 1}: {format(playlist.date_created, 'Do MMM YYYY')}</div>
       <p>Energy: {!playlist.energy ? '' : playlist.energy*10}</p>
       <p>Happiness: {!playlist.valence ? '' : playlist.valence*10}</p>
       <p>Tempo: {!playlist.tempo ? '' : playlist.tempo*10}</p>
       <p>Popularity: {!playlist.popularity ? '' : playlist.popularity/10}</p>
       <Playlist playlistId={playlist.playlist_id}/>
       <button className="btn mood" onClick={() => {
         TokenService.clearGenreToken();
         TokenService.clearPlaylistToken();
         this.props.history.push('/getWeather');
       }}>Make Another Playlist</button>
     </div>
    );
  });
  
  /*eslint-disable-next-line*/
  if (this.state.userPlaylists == 0 || this.state.userPlaylists == undefined ) {
    return (
    <div>
      <h4>No past playlists to display</h4>
      <button className="btn mood"><Link className="link" to="/getWeather">Make Another Playlist</Link></button>
    </div>
      )
  }
  return (
    <div>
      {mappedPlaylists}
    </div>
  );
}
}

export default UserPlaylists;