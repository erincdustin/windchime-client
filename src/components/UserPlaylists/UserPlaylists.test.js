import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom'
import { library } from '@fortawesome/fontawesome-svg-core'
import { faWind } from '@fortawesome/free-solid-svg-icons'
import UserPlaylists from './UserPlaylists';

library.add(faWind)

describe(`UserPlaylists`, () => {
  const props = {
    id: '123'
  }

  it('renders as expected', () => {
    const tree = renderer
      .create(<BrowserRouter>
          <UserPlaylists
            id={props.id}
            />
        </BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
})