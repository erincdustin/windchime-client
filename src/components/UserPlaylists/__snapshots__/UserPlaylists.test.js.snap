// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserPlaylists renders as expected 1`] = `
<div>
  <h4>
    No past playlists to display
  </h4>
  <button
    className="btn mood"
  >
    <a
      className="link"
      href="/getWeather"
      onClick={[Function]}
    >
      Make Another Playlist
    </a>
  </button>
</div>
`;
